### PR TITLE
indirect declare target test

### PR DESCRIPTION
**Pull request by mjcarr458**
Start of working on indirect declare target test. Here I copied down the basic structure provided by Swaroop in slack, but I still have a couple questions regarding it. Most importantly it uses the enter clause which was noted as a 5.2 feature. Since this is a 5.1 test is the a different work around I should use. Any suggestions or help here would be great appreciated.

### CODE REVIEW

**Comment by spophale** [comment]
@mjcarr458  - See my edits.

File: tests/5.1/target/test_target_declare_indirect.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+int fun2(){
+	return 10;
+}
+
+int fun3(){
+	return 15;
+}
+#pragma omp begin declare enter(fun1, fun2, fun3) indirect
```
Comment (tob2) [inline tests/5.1/target/test_target_declare_indirect.c:20]:
```suggestion
#pragma omp declare target enter(fun1, fun2, fun3) indirect
```
(or 'begin/end declare target' around the functions but then without 'enter' clause)

**Review by spophale — APPROVED** [review]

**Comment by fel-cab** [comment]
Is it necessary to have 3 functions and a rand to test the indirect?. I think the rand creates a not necessary complexity. I would change it to use the three functions and increment the values, instead of choosing one of them with a rand.

**Comment by mjcarr458** [comment]
Change enter() to to()

**Review by fel-cab — APPROVED** [review]

**Pull request merged**